Source/Core/Core/HW/SI.cpp

```diff
@@ -510,6 +510,16 @@ void ChangeDevice(SIDevices device, int channel)
 	}
 }
 
+void ChangeDeviceDeterministic(SIDevices device, int channel)
+{
+	// Called from savestates, so it needs to make it available immediately.
+	if (GetDeviceType(channel) != device)
+	{
+		CoreTiming::ScheduleEvent_Immediate(changeDevice, ((u64)channel << 32) | SIDEVICE_NONE);
+		CoreTiming::ScheduleEvent_Immediate(changeDevice, ((u64)channel << 32) | device);
```

Comment:
You still want to do this 5 billion cycles in the future, otherwise the game won't notice the controller is missing.